### PR TITLE
grt: ignore special nets when detecting dirty nets

### PR DESCRIPTION
**Pull request by eder-matheus**
Fixes [OR-1027](https://openroad.atlassian.net/browse/OR-1027).

### CODE REVIEW

**Review by osamahammad21 — CHANGES_REQUESTED** [review]
Thanks @eder-matheus that solves issue [OR-1027](https://openroad.atlassian.net/browse/OR-1027). However, I don't know why you need the extra nets_with_violations_ vector. It only holds the keys of _antennaViolations. The only change that helps here is checking if the net is special in inDbPostMoveInst callback.

**Comment by eder-matheus** [comment]
> Thanks @eder-matheus that solves issue [OR-1027](https://openroad.atlassian.net/browse/OR-1027). However, I don't know why you need the extra nets_with_violations_ vector. It only holds the keys of _antennaViolations. The only change that helps here is checking if the net is special in inDbPostMoveInst callback.

I've changed it trying to clarify that portion of code, but it is consuming unnecessary memory. I've just reverted it.

**Review by maliberty — APPROVED** [review]

File: src/grt/src/AntennaRepair.cpp [hunk 1]
```diff
@@ -326,7 +326,7 @@ AntennaCbk::AntennaCbk(GlobalRouter* grouter) : _grouter(grouter)
 void AntennaCbk::inDbPostMoveInst(odb::dbInst* inst)
 {
   for (odb::dbITerm* iterm : inst->getITerms()) {
-    if (iterm->getNet() != nullptr)
+    if (iterm->getNet() != nullptr && !iterm->getNet()->isSpecial())
       _grouter->addDirtyNet(iterm->getNet());
```
Comment (maliberty) [inline src/grt/src/AntennaRepair.cpp:330]: You could factor this to just call iterm->getNet() once.

**Review by osamahammad21 — APPROVED** [review]

**Comment by maliberty** [comment]
@osamahammad21 I think you'll need to resolve your "requested changes" so Eder can merge.

**Comment by osamahammad21** [comment]
> @osamahammad21 I think you'll need to resolve your "requested changes" so Eder can merge.

Not really. It just needs to go through the CI

**Pull request merged**